syntax = "proto3";

package runner;

service PlaybookService {
    rpc RunPlaybook (RunPlaybookRequest) returns (RunPlaybookResponse);
    rpc GetTaskStatus (PlaybookStatusRequest) returns (PlaybookStatusResponse);
}

message RunPlaybookRequest {
    string playbook_name = 1;   // Name of the playbook file
    string ssh_address = 2;     // root@172.0.0.1:22
    string ssh_password = 3;
    map<string, string> extra_vars = 4;
}

message RunPlaybookResponse{
    string id = 1;
    string status = 2;
}

message PlaybookStatusRequest{
    string id = 1;
}

message PlaybookStatusResponse {
    string status = 1;    // Status of the deployment
    string error = 2;     // Error message, if any
}
